src/fparser/common/readfortran.py

```diff
@@ -1047,6 +1077,28 @@ def warning(self, message, item=None):
 
     # Auxiliary methods for processing raw source lines:
 
+    def handle_cpp_directive(self, line):
+        """Determine whether the current line is likely to hold
+        C preprocessor directive.
+
+        If the first character of a line is the symbol ``#`` it is assumed
```

Comment:
This statement seems to be at odds with the implementation, which strips any white space. One of the tests also tests that a CPP directive is recognised even if there is leading white space.

Reply:
Mmm, have forgotten to change that after adding the `lstrip()`. Thanks.